packages/core/src/util/resolvers.ts

```diff
@@ -155,36 +156,10 @@ export const resolveSchema = (
   // with absolute paths here, so that we don't need to keep two different
   // schemas around
   if (resultSchema !== undefined && resultSchema.$ref !== undefined) {
-    try {
-      return retrieveResolvableSchema(schema, resultSchema.$ref);
-    } catch (e) {
-      return retrieveResolvableSchema(rootSchema, resultSchema.$ref);
-    }
+    return resolveSchema(schema, resultSchema.$ref, rootSchema)
```

Comment:
Is there an use case of trying to resolve the `$ref` against the `schema` first instead of directly going to the `rootSchema`?

Reply:
No, we can directly resolve against the rootSchema here, since we tried resolving against schema beforehand.